xds/src/test/java/io/grpc/xds/ServerXdsClientTest.java

```diff
@@ -236,20 +227,6 @@ private static DiscoveryRequest buildDiscoveryRequest(
         .build();
   }
 
-  /** Error when 2 ListenerWatchers registered. */
```

Comment:
This test fails because the checks were inside the Runnable. If the checks were outside this test would work. Even if the test is trivial why remove it?

Reply:
This test doesn't make sense to me. It doesn't test anything. But if you want, I can just put it back and also let the check run outside the SynchronizationContext. It's unclear to me how you make synchronization work for server side XdsClient.

Reply:
It doesn't make sense to you - alright. But unless you think it is asserting the wrong thing let's leave it there. I have been burnt by removed tests so ...

Also like I mentioned in the main code, the checks on the input arguments (these are not checks on the states) should *not* be inside the scheduled tasks. Even if you say that check on the existing `listenerWatcher` is a check on the state failing when it is non-null is failing early which is okay.

Reply:
Alright. I moved the checks outside the sync context and put back the test case.